docs/standard/parallel-programming/snippets/cs/cancellation1.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class CancellationExample
+{
+    public static async Task Main()
+    {
+        var random = new Random();
+        using var cts = new CancellationTokenSource();
+        CancellationToken token = cts.Token;
+        var timer = new Timer(Elapsed, cts, 5000, Timeout.Infinite);
+
+        var task = Task.Run(
+            () =>
+            {
+                var product33 = new List<int>();
+                for (int ctr = 1; ctr < short.MaxValue; ctr++)
+                {
+                    if (token.IsCancellationRequested)
+                    {
+                        Console.WriteLine("\nCancellation requested in antecedent...\n");
+                        token.ThrowIfCancellationRequested();
+                    }
+                    if (ctr % 2000 == 0)
+                    {
+                        int delay = random.Next(16, 501);
+                        Thread.Sleep(delay);
```

Comment:
I *think* all these work if you replace `Thread.Sleep` with `await Task.Delay(delay)`, and that's a pattern we should be showing instead.